samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```
@@ -209,20 +211,23 @@ public String getJobModelVersion() {
    * @param oldVersion - used to validate, that no one has changed the version in the meanwhile.
    * @param newVersion - new version.
    */
-  public void publishNewJobModelVersion(String oldVersion, String newVersion) {
+  public void publishJobModelVersion(String oldVersion, String newVersion) {
     Stat stat = new Stat();
     String currentVersion = zkClient.<String>readData(keyBuilder.getJobModelVersionPath(), stat);
-    LOG.info("pid=" + processorId + " publishing new version: " + newVersion + "; oldVersion = " + oldVersion + "(" + stat.getVersion() + ")");
+    LOG.info("pid=" + processorId + " publishing new version: " + newVersion + "; oldVersion = " + oldVersion + "(" + stat
+            .getVersion() + ")");
 
     if (currentVersion != null && !currentVersion.equals(oldVersion)) {
-      throw new SamzaException("Someone change JMVersion while Leader was generating: expected" + oldVersion + ", got " + currentVersion);
+      throw new SamzaException(
+          "Someone change JobModelVersion while the leader was generating one: expected" + oldVersion + ", got " + currentVersion);
     }
     // data version is the ZK version of the data from the ZK.
     int dataVersion = stat.getVersion();
-    stat = zkClient.writeDataReturnStat(keyBuilder.getJobModelVersionPath(), newVersion, dataVersion);
-    if (stat.getVersion() != dataVersion + 1)
-      throw new SamzaException("Someone changed data version of the JMVersion while Leader was generating a new one. current= " + dataVersion + ", old version = " + stat.getVersion());
-
+    try {
+      stat = zkClient.writeDataReturnStat(keyBuilder.getJobModelVersionPath(), newVersion, dataVersion);
+    } catch (Exception e) {
+      LOG.error("publish job model version failed for new version = " + newVersion + "; old version = " + oldVersion, e);
```

Shouldn't we re-throw the exception here?